doc/configuration.md

```diff
@@ -1,94 +1,237 @@
+  * **units** (list of objects): the list of systemd units.
+    * **name** (string, required): the name of the unit. This must be suffixed with a valid unit type (e.g. "thing.service").
+    * **enable** (boolean): whether or not the service shall be enabled. When true, the service is enabled. In order for this to have any effect, the unit must have an install section.
+    * **mask** (boolean): whether or not the service shall be masked. When true, the service is masked by symlinking it to `/dev/null`.
+    * **contents** (string): the contents of the unit.
+    * **dropins** (list of objects): the list of drop-ins for the unit.
+      * **name** (string, required): the name of the drop-in. This must be suffixed with ".conf".
+      * **contents** (string): the contents of the drop-in.
+* **networkd** (object): describes the desired state of the networkd files.
+  * **units** (list of objects): the list of networkd files.
+    * **name** (string, required): the name of the file. This must be suffixed with a valid unit type (e.g. "00-eth0.network").
+    * **contents** (string): the contents of the networkd file.
+* **passwd** (object): describes the desired additions to the passwd database.
+  * **users** (list of objects): the list of accounts to be added.
+    * **name** (string, required): the username for the account.
+    * **password_hash** (string): the encrypted password for the account.
+    * **ssh_authorized_keys** (list of strings): a list of SSH keys to be added to the user's authorized_keys.
+    * **create** (object): contains the set of options to be used when creating the user. A non-null entry indicates that the user account shall be created.
+      * **uid** (integer): the user ID of the new account.
+      * **gecos** (string): the GECOS field of the new account.
+      * **home_dir** (string): the home directory of the new account.
+      * **no_create_home** (boolean): whether or not to create the user's home directory.
+      * **primary_group** (string): the name or ID of the primary group of the new account.
+      * **groups** (list of strings): the list of supplementary groups of the new account.
+      * **no_user_group** (boolean): whether or not to create a group with the same name as the user.
+      * **no_log_init** (boolean): whether or not to add the user to the lastlog and faillog databases.
+      * **shell** (string): the login shell of the new account.
+  * **groups** (list of objects): the list of groups to be added.
+    * **name** (string, required): the name of the group.
+    * **gid** (integer): the group ID of the new group.
+    * **password_hash** (string): the encrypted password of the new group.
+* **etcd**
+  * **version** (string, required): the version of etcd to be run
+  * **_other options_** (string): this section accepts any valid etcd options for the version of etcd specified. For a comprehensive list, please consult etcd's documentation
```

Comment:
Should we indicate options should be in snake_case and not spine-case like the cli args are?